cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -556,3 +557,57 @@ func restoreEnv(e map[string]string) {
 		os.Setenv(k, v)
 	}
 }
+
+func TestKubeletVersionCheck(t *testing.T) {
+	type T struct {
+		kubeletVersion string
+		k8sVersion     string
+		expectErrors   bool
+		expectWarnings bool
+	}
+
+	cases := []T{
+		{"v1.7.3", "v1.7.8", true, false},          // Too old kubelet
+		{"v1.9.0", "v1.9.5", false, false},         // kubelet within same major.minor as control plane
+		{"v1.9.0", "v1.10.1", false, false},        // kubelet is lower than control plane, but newer than minimally supported
+		{"v1.10.0-alpha.1", "v1.9.1", false, true}, // kubelet is newer than control plane
```

Comment:
I would expect this to fail

Reply:
Also add a proper `v1.10.3` and `v1.9.4` test that should fail

Reply:
Using development builds of kubelet on top of stable control plane would be good to allow. See other comment.

Reply:
ok